### PR TITLE
Update packages

### LINKED ISSUE #17482: Update external dependencies to latest GA releases
- ~~System.Threading.Tasks.Extensions (4.5.2)~~
https://github.com/aspnet/EntityFrameworkCore/blob/release/3.0/src/EFCore/EFCore.csproj#L27

- Castle.Core (4.2.1)
https://github.com/aspnet/EntityFrameworkCore/blob/release/3.0/src/EFCore.Proxies/EFCore.Proxies.csproj#L19

- Microsoft.Azure.Cosmos (3.0.0)
https://github.com/aspnet/EntityFrameworkCore/blob/release/3.0/src/EFCore.Cosmos/EFCore.Cosmos.csproj#L25

- SQLitePCLRaw (2.0.1)

### LINKED ISSUE #17625: Update EFCore.Analyzer
Two concerns:
1. Can we update Microsoft.CodeAnalysis.CSharp.Workspaces? Will this break compatibility with certain SDK/VS versions? How high can we go? (EF Core 3 requires at least VS 16.3 or the 3.0 SDK)
2. Newer packages will require us to change from .NET Standard 1.3 to 2.0. This isn't actually breaking, is it?

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #17482, fixes #17625

### CODE REVIEW

File: src/EFCore.Analyzers/InternalUsageDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -45,6 +45,7 @@ public override void Initialize(AnalysisContext context)
                 SyntaxKind.SimpleMemberAccessExpression,
                 SyntaxKind.ObjectCreationExpression,
                 SyntaxKind.ClassDeclaration);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
```
Comment (bricelam) [inline src/EFCore.Analyzers/InternalUsageDiagnosticAnalyzer.cs]: @roji I don't know what constitutes "generated code", but it doesn't sound important to run our analyzer there, right?

File: eng/Versions.props [hunk 2]
```diff
@@ -10,18 +10,18 @@
...
     <NetTopologySuiteIOSpatiaLitePackageVersion>2.0.0</NetTopologySuiteIOSpatiaLitePackageVersion>
     <NetTopologySuiteIOSqlServerBytesPackageVersion>2.0.0</NetTopologySuiteIOSqlServerBytesPackageVersion>
-    <SQLitePCLRawBundleESqlite3PackageVersion>2.0.0</SQLitePCLRawBundleESqlite3PackageVersion>
-    <SQLitePCLRawBundleESqlcipherPackageVersion>2.0.0</SQLitePCLRawBundleESqlcipherPackageVersion>
-    <SQLitePCLRawCorePackageVersion>2.0.0</SQLitePCLRawCorePackageVersion>
-    <StyleCopAnalyzersPackageVersion>1.1.1-beta.61</StyleCopAnalyzersPackageVersion>
+    <SQLitePCLRawBundleESqlite3PackageVersion>2.0.1</SQLitePCLRawBundleESqlite3PackageVersion>
+    <SQLitePCLRawBundleESqlcipherPackageVersion>2.0.1</SQLitePCLRawBundleESqlcipherPackageVersion>
+    <SQLitePCLRawCorePackageVersion>2.0.1</SQLitePCLRawCorePackageVersion>
+    <StyleCopAnalyzersPackageVersion>1.1.118</StyleCopAnalyzersPackageVersion>
```
Comment (dougbu) [inline eng/Versions.props]: Please confirm all of these new versions are OSS-approved. @Pilchie will be back on Monday and @divega may have the info today if necessary.

File: src/EFCore.Analyzers/InternalUsageDiagnosticAnalyzer.cs (hunk 1 above)
Comment (roji) [reply src/EFCore.Analyzers/InternalUsageDiagnosticAnalyzer.cs]: Wasn't aware of this... Any idea what constitutes generated code for roslyn?

In any case, I agree it doesn't sound important, but shouldn't we leave whatever the default behavior is? In other words, are all analyzers now supposed to explicitly disable themselves for generated code, and if not, how are we special? Also are there any actual issues with leaving the default behavior?

Comment (bricelam) [reply src/EFCore.Analyzers/InternalUsageDiagnosticAnalyzer.cs]: I could just suppress the warning. I think the idea behind the analyses to to make more analyzers think about it and be explicit. They may be thinking about changing the default in a future release

Comment (roji) [reply src/EFCore.Analyzers/InternalUsageDiagnosticAnalyzer.cs]: Oh, didn't realize this was an analyzer warning. I guess it makes sense to do what it says then...

**Comment by bricelam** [comment]
@AndriySvyryd It failed again.
```
System.InvalidOperationException: Delete database failed. Status code: Unauthorized. Message: 'The input authorization token can't serve the request. Please check that the expected payload is built as per the protocol, and check the key being used. Server used the following payload to sign: 'gettue, 24 sep 2019 16:01:24 gmt'ActivityId: 20370510-e924-4a37-9a37-6ca2becd62e9, Microsoft.Azure.Documents.Common/2.5.1, Darwin/10.13 cosmos-netstandard-sdk/3.2.1'
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.DeleteDatabaseOnceAsync(DbContext _, Object __, CancellationToken cancellationToken) in /_/src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs:line 194
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.DeleteDatabaseOnce(DbContext context, Object state) in /_/src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs:line 160
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementation[TState,TResult](Func`3 operation, Func`3 verifySucceeded, TState state) in /_/src/EFCore/Storage/ExecutionStrategy.cs:line 173
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded) in /_/src/EFCore/Storage/ExecutionStrategy.cs:line 159
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.DeleteDatabase() in /_/src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs:line 149
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosDatabaseCreator.EnsureDeleted() in /_/src/EFCore.Cosmos/Storage/Internal/CosmosDatabaseCreator.cs:line 128
   at Microsoft.EntityFrameworkCore.Infrastructure.DatabaseFacade.EnsureDeleted() in /_/src/EFCore/Infrastructure/DatabaseFacade.cs:line 91
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosTestStore.Clean(DbContext context) in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosTestStore.cs:line 146
   at Microsoft.EntityFrameworkCore.TestUtilities.TestStore.Initialize(Func`1 createContext, Action`1 seed, Action`1 clean) in /_/test/EFCore.Specification.Tests/TestUtilities/TestStore.cs:line 78
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosTestStore.Initialize(Func`1 createContext, Action`1 seed, Action`1 clean) in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosTestStore.cs:line 81
   at Microsoft.EntityFrameworkCore.TestUtilities.TestStore.Initialize(IServiceProvider serviceProvider, Func`1 createContext, Action`1 seed, Action`1 clean) in /_/test/EFCore.Specification.Tests/TestUtilities/TestStore.cs:line 48
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosTestStore.CreateInitialized(String name, Action`1 extensionConfiguration) in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosTestStore.cs:line 27
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosDbConfiguredConditionAttribute.TryConnectAsync() in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs:line 34
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosDbConfiguredConditionAttribute.TryConnectAsync() in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs:line 60
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosDbConfiguredConditionAttribute.IsMetAsync() in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs:line 23
   at Microsoft.EntityFrameworkCore.TestUtilities.Xunit.XunitTestCaseExtensions.TrySkipAsync(XunitTestCase testCase, IMessageBus messageBus) in /_/test/EFCore.Specification.Tests/TestUtilities/Xunit/XunitTestCaseExtensions.cs:line 43
   at Microsoft.EntityFrameworkCore.TestUtilities.Xunit.ConditionalFactTestCase.RunAsync(IMessageSink diagnosticMessageSink, IMessageBus messageBus, Object[] constructorArguments, ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource) in /_/test/EFCore.Specification.Tests/TestUtilities/Xunit/ConditionalFactTestCase.cs:line 35
   at Xunit.Sdk.TestMethodRunner`1.RunTestCasesAsync() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestMethodRunner.cs:line 136
   at Xunit.Sdk.TestMethodRunner`1.RunAsync() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestMethodRunner.cs:line 106
   at Xunit.Sdk.TestClassRunner`1.RunTestMethodsAsync() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestClassRunner.cs:line 213
   at Xunit.Sdk.TestClassRunner`1.RunAsync() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestClassRunner.cs:line 171
   at Xunit.Sdk.TestCollectionRunner`1.RunTestClassesAsync() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestCollectionRunner.cs:line 130
   at Xunit.Sdk.TestCollectionRunner`1.RunAsync() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestCollectionRunner.cs:line 101
   at Xunit.Sdk.TestAssemblyRunner`1.RunTestCollectionsAsync(IMessageBus messageBus, CancellationTokenSource cancellationTokenSource) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestAssemblyRunner.cs:line 231
   at Xunit.Sdk.XunitTestAssemblyRunner.RunTestCollectionsAsync(IMessageBus messageBus, CancellationTokenSource cancellationTokenSource) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\XunitTestAssemblyRunner.cs:line 165
   at Xunit.Sdk.TestAssemblyRunner`1.RunAsync() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestAssemblyRunner.cs:line 196
   at Xunit.Sdk.XunitTestFrameworkExecutor.RunTestCases(IEnumerable`1 testCases, IMessageSink executionMessageSink, ITestFrameworkExecutionOptions executionOptions) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\XunitTestFrameworkExecutor.cs:line 95
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

**Comment by bricelam** [comment]
Version 3.1.1 of Microsoft.Azure.Cosmos seems to work. @AndriySvyryd do you wan to investigate and submit an issue? It only seems to affect Linux and macOS

**Comment by smitpatel** [comment]
@bricelam - Let's file an issue. (@AndriySvyryd  is on vacation)

**Comment by bricelam** [comment]
Filed Azure/azure-cosmos-dotnet-v3#847

**Comment by AndriySvyryd** [comment]
My commit is in, please rebase

**Pull request merged**